spec/integration/account_migration_spec.rb

```diff
@@ -0,0 +1,204 @@
+require "integration/federation/federation_helper"
+
+def create_remote_contact(user, pod_host)
+  FactoryGirl.create(
+    :contact,
+    user:   user,
+    person: FactoryGirl.create(
+      :person,
+      pod:             Pod.find_or_create_by(url: "http://#{pod_host}"),
+      diaspora_handle: "#{r_str}@#{pod_host}"
+    )
+  )
+end
+
+shared_examples_for "old person account is closed and profile is cleared" do
+  subject { old_user.person }
+
+  before do
+    run_migration
+    subject.reload
+  end
+
+  include_examples "it makes account closed and clears profile"
+end
+
+shared_examples_for "old person doesn't have any reference left" do
+  let(:person) { old_user.person }
+
+  before do
+    DataGenerator.create(person, :generic_person_data)
+  end
+
+  def account_removal_method
+    run_migration
+    person.reload
+  end
+
+  include_examples "it removes the person associations"
+
+  include_examples "it removes the person conversations"
+end
+
+shared_examples_for "every migration scenario" do
+  it_behaves_like "it updates person references"
+
+  it_behaves_like "old person account is closed and profile is cleared"
+
+  it_behaves_like "old person doesn't have any reference left"
+end
+
+shared_examples_for "migration scenarios with local old user" do
+  it "locks the old user account" do
+    run_migration
+    expect(old_user.reload).to be_a_locked_account
+  end
+end
+
+shared_examples_for "migration scenarios initiated remotely" do
+  it "resends known contacts to the new user" do
+    contacts = Array.new(2) { FactoryGirl.create(:contact, person: old_user.person, sharing: true) }
+    expect(DiasporaFederation::Federation::Sender).to receive(:private)
+      .twice do |sender_id, obj_str, _urls, _xml|
+      expect(sender_id).to eq(contacts.first.user_id)
+      expect(obj_str).to eq("Contact:#{contacts.first.user.diaspora_handle}:#{new_user.diaspora_handle}")
+      contacts.shift
+      []
+    end
+    inlined_jobs { run_migration }
+  end
+end
+
+shared_examples_for "migration scenarios initiated locally" do
+  it "dispatches account migration message to the federation" do
+    expect(DiasporaFederation::Federation::Sender).to receive(:public) do |sender_id, obj_str, urls, xml|
+      if old_user.person.remote?
+        expect(sender_id).to be_nil
```

Comment:
Didn't realize that while reading the code, but reading the test makes it clearer: sender_id is only used for logging, so nil is OK, but how about using the old diaspora ID here to have a more useful logging?